docs/multi_contact_migration.md

```diff
@@ -0,0 +1,382 @@
+# Multi-Contact Support Migration Guide
+
+## Overview
+
+This migration adds support for multiple contact email addresses per host, replacing the single `contact` field with a many-to-many relationship to a new `HostContact` model.
+
+## Changes Made
+
+### 1. New Model: `HostContact`
+
+- Location: `mreg/models/host.py`
+- Purpose: Store individual contact email addresses that can be shared across multiple hosts
+- Fields:
+  - `id`: Primary key
+  - `email`: EmailField - the contact email address
+  - `created_at`: Timestamp
+  - `updated_at`: Timestamp
+
+### 2. Updated `Host` Model
+
+- Added `contacts` ManyToManyField relationship to `HostContact`
+- Removed deprecated `contact` field
+- Added helper methods:
+  - `add_contact(email: str)`: Add a contact email to the host
+  - `remove_contact(email: str)`: Remove a contact email from the host
+  - `get_contact_emails()`: Get list of all contact emails
+
+### 3. Migration
+
+#### Migration 0016: `0016_host_contacts.py`
+
+This single migration performs all the necessary changes:
+
+- Creates the `HostContact` model
+- Adds the `contacts` ManyToMany field to `Host`
+- Migrates existing data from `Host.contact` to `HostContact` instances
+- Removes the deprecated `contact` field
+- **Reversible**: Can migrate back by copying the first contact to the old field
+
+### 4. API Changes
+
+#### Serializers (`mreg/api/v1/serializers.py`)
+
+- Added `HostContactSerializer` for representing contact objects
+- Updated `HostSerializer`:
+  - Added `contacts` field (read-only) - returns list of `HostContact` objects with full details (id, email, timestamps)
+  - Added `contact_emails` field (write-only) - accepts a list of email strings for creating/updating contacts
```

Comment:
Is there a reason for this separation of read-only and write-only fields? Why can't this field accept a list of emails in POST/PUT/PATCH requests and still return the full email objects in GET requests?

It seems to differ from fields in other models, where we read from and write to the same field.

Reply:
The main reason is that `contacts` would then deliver one thing (objects) and receive another (emails). It's fully doable, like we do elsewhere, but it feels a bit weird as most other places have at least more than one field that we actually use. I went back and forth on this a few times before going for the split.

I think you're right though, I'll merge them.